tests/models/roformer/test_modeling_roformer.py

```diff
@@ -220,6 +220,30 @@ def create_and_check_for_causal_lm(
         result = model(input_ids, attention_mask=input_mask, token_type_ids=token_type_ids, labels=token_labels)
         self.parent.assertEqual(result.logits.shape, (self.batch_size, self.seq_length, self.vocab_size))
 
+    def create_and_check_for_generate_causal_lm(
+        self,
+        config,
+        input_ids,
+        token_type_ids,
+        input_mask,
+        sequence_labels,
+        token_labels,
+        choice_labels,
+    ):
+        torch.manual_seed(0)
+        config.is_decoder = True
+        config.bos_token_id = 0
+        config.eos_token_id = 1
+        config.pad_token_id = 2
+        model = RoFormerForCausalLM(config=config)
+        model.to(torch_device)
+        model.eval()
+        with torch.no_grad():
+            input_ids = model.generate(do_sample=False, max_length=10)
+            output = model(input_ids=input_ids)
```

Comment:
When using the `generate()` method to generate text token by token, a unique sequence is obtained when using greedy decoding (`do_sample=False`). If the complete sequence is input to the model (`output = model(input_ids=input_ids)`), the resulting output should be consistent with the output generated by feeding the input text token by token, except for the `bos_token_id` added by `generate()` at the beginning and the last token generated by the complete sequence.

Reply:
Thanks for clarifying! Can you run the styling checks? `make fixup`

Reply:
Actually there is a test in transformers that ensures that the output with/without past are the same. I am wondering why it was not triggered. `test_attention_outputs` should be doing this. 
Otherwise we usually have this test implemented as: (example with T5)
```python 
    def create_and_check_generate_with_past_key_values(
        self,
        config,
        input_ids,
        decoder_input_ids,
        attention_mask,
        decoder_attention_mask,
        lm_labels,
    ):
        model = T5ForConditionalGeneration(config=config).to(torch_device).eval()
        torch.manual_seed(0)
        output_without_past_cache = model.generate(
            input_ids[:1], num_beams=2, max_length=5, do_sample=True, use_cache=False
        )
        torch.manual_seed(0)
        output_with_past_cache = model.generate(input_ids[:1], num_beams=2, max_length=5, do_sample=True)
        self.parent.assertTrue(torch.all(output_with_past_cache == output_without_past_cache))
```

Reply:
> Thanks for clarifying! Can you run the styling checks? `make fixup`

Thank you, `make fixup` version has been submitted

Reply:
> Actually there is a test in transformers that ensures that the output with/without past are the same. I am wondering why it was not triggered. `test_attention_outputs` should be doing this. Otherwise we usually have this test implemented as: (example with T5)
> 
> ```python
>     def create_and_check_generate_with_past_key_values(
>         self,
>         config,
>         input_ids,
>         decoder_input_ids,
>         attention_mask,
>         decoder_attention_mask,
>         lm_labels,
>     ):
>         model = T5ForConditionalGeneration(config=config).to(torch_device).eval()
>         torch.manual_seed(0)
>         output_without_past_cache = model.generate(
>             input_ids[:1], num_beams=2, max_length=5, do_sample=True, use_cache=False
>         )
>         torch.manual_seed(0)
>         output_with_past_cache = model.generate(input_ids[:1], num_beams=2, max_length=5, do_sample=True)
>         self.parent.assertTrue(torch.all(output_with_past_cache == output_without_past_cache))
> ```

Maybe the pre-training model released by roformer is mainly mask language model, test case does not take causal language model into account

Reply:
After reading @ArthurZucker 's comment, I think it makes sense to add a test similar to what he suggested, what do you think about replacing the test you have implemented with the one suggested by Arthur?

Reply:
Not recommended by the author, but I wrote it based on the test cases of gpt2, which uses the predefined `expected_ output_ Ids`, but due to the lack of a suitable `RoFormerForCausalLM` pre training model, it is difficult to define `expected_ output_ Ids`, so I use the sequence generated by `generate()`

```
@require_torch
class GPT2ModelLanguageGenerationTest(unittest.TestCase):
    def _test_lm_generate_gpt2_helper(
        self,
        gradient_checkpointing=False,
        reorder_and_upcast_attn=False,
        scale_attn_by_inverse_layer_idx=False,
        verify_outputs=True,
    ):
        model = GPT2LMHeadModel.from_pretrained(
            "gpt2",
            reorder_and_upcast_attn=reorder_and_upcast_attn,
            scale_attn_by_inverse_layer_idx=scale_attn_by_inverse_layer_idx,
        )
        if gradient_checkpointing:
            model.gradient_checkpointing_enable()
        else:
            model.gradient_checkpointing_disable()
        model.to(torch_device)

        # The dog
        input_ids = torch.tensor([[464, 3290]], dtype=torch.long, device=torch_device)

        # The dog was found in a field near the intersection of West and West Streets.\n\nThe dog
        # fmt: off
        expected_output_ids = [
            464, 3290, 373, 1043, 287, 257, 2214, 1474, 262, 16246, 286, 2688, 290, 2688, 27262, 13, 198, 198, 464, 3290,
        ]
        # fmt: on
        output_ids = model.generate(input_ids, do_sample=False)
        if verify_outputs:
            self.assertListEqual(output_ids[0].tolist(), expected_output_ids)
```